tests/test_prepayment_before_generation.py

```
@@ -115,3 +117,35 @@ def test_payment_confirmation_guard_covers_every_menu_document() -> None:
         )
         assert blocked.allowed is False, kind
         assert allowed.allowed is True, kind
+
+
+def test_consumed_prepayment_cannot_reenter_generation(monkeypatch) -> None:
+    class State:
+        async def get_data(self):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `get_data`

(ANN202)